### ISSUE TITLE
Tagging and deleting

### ISSUE THREAD

**Issue by jesta030** [bpatrik/pigallery2#183]
**Is your feature request related to a problem? Please describe.**
I have been looking everywhere for a photo viewing/organizing app that doesn't mess with the files and directory structure or stores information like tags in a database that is inaccessible to other apps. I have only found other people looking for the same functionality but no viable solution. PiGallery2 is awesome in it's simplicity and work with IPTC tags and it would be perfect if it could edit those tags from the WebUI and delete unwanted pictures.

**Describe the solution you'd like**
A simple interface to add tags. Bonus if it suggests the most recently used tags so they don't have to be entered every time.
A trash can button to quickly delete pictures.

**Describe alternatives you've considered** (optional)
I am currently using windows explorer to tag pictures which works but is cumbersome and rather slow. I have used Synology's Photo Station before which has tagging and deleting features but is only available on Synology DSM or in Xpenology which is not really stable or supported and doesn't like external storage.

Any chance of these features finding their way into PiGallery2?

P.S. I'd like to contribute a german localization, how do I go about doing this?

**Comment by palitu** [comment]
@jesta030 ,

I agree!  i haven't found a better option, and want the exact same functionality.

I use Digikam for tagging, which i would imagine is better than windows explorer.

**Comment by bpatrik** [comment]
Hi,

This is related to #38 #76 

In short I'm still not sure that I want to support photo manipulation at the current form of the app. I would like to keep the photos folder read only. I'm sorry.

To translate the app the easiest way probably to clone the `messages.en.xlf` to `message.de.xlf`: https://github.com/bpatrik/pigallery2/blob/master/src/frontend/translate/messages.en.xlf
And translate the `target` tags in them. You can find example of translations in the same folder. 

you can also test if you want:
-download the source and build it when you ready with the translation.
But you can save the time for that. I'll accept the PR if you promise me that you do not mess up the `<>` tags :)

**Comment by 02015678** [comment]
I think there is a Github repo for reference.
https://github.com/Webreaper/Damselfly

It is possibly still in early stage, since it has lower forks and stars than this project.
But it does provides add/edit/remove EXIF/IPTC tag feature, which is very useful to organize photo.

It not only provides an unified web user interface for across PC/Mac/Mobile devices, but also provides native client for Windows, Mac or Linux desktop.

While on the other hand, Damselfly is NOT as light-weight as PiGallery2. 
PiGallery2 docker image is about 245MB, plus additional MariaDB image 115MB, in total 360MB.
While Damselfly is about 1.1GB, which is even larger than Photoprism 630MB.
That is possibly why PiGallery2 keeps working in read-only fashion, at least for now.

I'll say I will have a try on Damselfly. But currently I am not sure whether it can run fast.
PhotoPrism speed is not good. And PiGallery2 is optimized to be blazing fast.

**Comment by pepa65** [comment]
PiGallery2 is fantastic, and does all I want, is configurable and fast. The only thing is, my wife really wants a delete option. Would it not be possible to keep the storage folders read-only by default, but for the people that need it to allow them to be to delete images??

**Comment by bpatrik** [comment]
FYI: with the latest extension feature, everyone can implement a simple button that appears an each image and you can use that do modify images on the server side. The sample extension implements a minimal deletion and edit logic.

See: https://github.com/bpatrik/pigallery2/issues/1032#issuecomment-3300380098

**Comment by bpatrik** [comment]
I will close this issue as with extension this behavior is NOW possible to implement.

**Comment by pepa65** [comment]
`this behavior is not possible to implement`
"not" => "now"..!
